### PR TITLE
Jordan/1982 session modal is a real modal

### LINKED ISSUE #1982: make session a modal
Description:

Should match the designs from Zplin, with no background-color: https://app.zeplin.io/project/59e66beff6721ea8bf562e5b/screen/5c376fc7200bd0298b169788

### PR DESCRIPTION
**Pull request by jbibla**
Closes #1982 

**Description:**

- made session a real modal
- this does leave a few usability issues to be addressed in future PRs

screen shot 2019-02-27 at 10 57 06 am

Thank you! 🚀

---

For contributor:

- [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by fedekunze** [comment]
> this does leave a few usability issues to be addressed in future PRs

can you elaborate ? just for curiosity

**Comment by fedekunze** [comment]
also some coverage is missing

**Comment by jbibla** [comment]
> can you elaborate ? just for curiosity

yes! mainly, we are not redirecting users quite as strongly anymore. so if you close the modal on the wallet page for example, it is just blank. in a new PR this should display a "you're not signed in" message instead of just being blank.

same applies to transactions, my delegations, etc..

File: test/unit/specs/components/wallet/TmModalSendConfirmation.spec.js [hunk 1]
```diff
@@ -1,50 +0,0 @@
-import { mount } from "@vue/test-utils"
-import TmModalSendConfirmation from "renderer/components/wallet/TmModalSendConfirmation"
-
-describe(`TmModalSendConfirmation`, () => {
-  let wrapper
-  const propsData = {
-    amount: 12345,
-    denom: `funkycoin`,
```
Comment (fedekunze) [inline test/unit/specs/components/wallet/TmModalSendConfirmation.spec.js]: Rip funkycoin

Comment (jbibla) [reply test/unit/specs/components/wallet/TmModalSendConfirmation.spec.js]: 🕺

File: app/src/renderer/components/common/TmSession.vue [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
-      v-else-if="session.modals.session.state == 'delete'"
-    />
-    <session-hardware v-else-if="session.modals.session.state == 'hardware'" />
-    <session-import v-else-if="session.modals.session.state == 'import'" />
+  <modal v-if="active" :close="close">
+    <div slot="main">
+      <session-welcome v-if="session.modals.session.state == 'welcome'" />
+      <session-sign-up v-else-if="session.modals.session.state == 'sign-up'" />
+      <session-sign-in v-else-if="session.modals.session.state == 'sign-in'" />
+      <session-account-delete
```
[the remark this answers is not in the document]
Comment (jbibla) [reply app/src/renderer/components/common/TmSession.vue]: not sure. what do you think?

File: app/src/renderer/components/common/TmPage.vue [hunk 3]
```diff
@@ -9,7 +9,7 @@
       </h3>
       <slot slot="menu-body" name="menu-body">
         <tm-balance v-if="session.signedIn" />
-        <tool-bar :refresh="refreshable" />
+        <tool-bar />
```
Comment (sabau) [inline app/src/renderer/components/common/TmPage.vue]: I think if we want to kill this feature you should also remove from all the pages where it's used + the function definitions.
Do we want to remove the refresh button?

Comment (sabau) [reply app/src/renderer/components/common/TmPage.vue]: Now that I think about it this should be a separate PR 🚀

Comment (sabau) [reply app/src/renderer/components/common/TmPage.vue]: Sorry I can't resist
oops

Comment (faboweb) [reply app/src/renderer/components/common/TmPage.vue]: 👎 please put the refresh back. as long as we don't trust the UI is updating everything nicely we should have this.

File: app/src/renderer/components/common/TmSession.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (faboweb) [reply app/src/renderer/components/common/TmSession.vue]: This is never showing. I think we should actually show it as currently a user (or dev) is not able to delete an account. but not a prio as this is only visible in experimental mode.

**Comment by fedekunze** [comment]
Also, when offline, clicking `Wallet`, `Transactions` or `My Delegations` and then signing in should redirect to those pages instead of going to the `Network` page

**Comment by fedekunze** [comment]
imo we should also add a close button on the individual modals (such as the` Import with seed`, `Hardware`, etc)

File: app/src/renderer/components/common/TmSession.vue [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
+        v-else-if="session.modals.session.state == 'delete'"
+      />
+      <session-hardware v-else-if="session.modals.session.state == 'hardware'" />
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmSession.vue]: can we rename this to `session-ledger`? and have something like
```suggestion
      <session-ledger v-else-if="session.modals.session.state == 'ledger'" />
```

**Comment by faboweb** [comment]
good findings @fedekunze

File: app/src/renderer/components/common/TmSession.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbibla) [reply app/src/renderer/components/common/TmSession.vue]: > experimental mode

lol. we opened a can of worms.

File: app/src/renderer/components/common/TmPage.vue (hunk 3 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmPage.vue]: > as long as we don't trust the UI is updating everything nicely we should have this.

if a UI team can't trust their UI — WHAT ARE WE EVEN DOING HERE

---

sorry, i will add the refresh back

File: app/src/renderer/components/common/TmSession.vue (hunk 4 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmSession.vue]: yeah that would be more descriptive indeed

File: app/src/renderer/components/common/TmSession.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (faboweb) [reply app/src/renderer/components/common/TmSession.vue]: #followup

File: app/src/renderer/components/common/TmPage.vue (hunk 3 above)
Comment (faboweb) [reply app/src/renderer/components/common/TmPage.vue]: we are scientists not believers ;)

**Comment by jbibla** [comment]
thank you @fedekunze - i opened an issue and already started working on it to address your first 3 points - #2121.

the fundraiser styles do seem to be broken, so i'll resolve that here 👍

**Comment by fedekunze** [comment]
Ok but imo point 1 should be addressed here since it always throws an error notification

**Comment by fedekunze** [comment]
Also:

- invalid password throws a notification (should be displayed on the modal) 

screen shot 2019-03-03 at 10 35 50 pm

- Signing in still redirects to home page (network page)

**Comment by faboweb** [comment]
fixed error on pagetransaction

**Comment by faboweb** [comment]
sign in error is now inline

File: app/src/renderer/components/common/TmSessionHardware.vue [hunk 5]
```diff
@@ -8,7 +8,9 @@
         <div class="tm-session-title">
           Sign In
         </div>
-        <a @click="help"> <i class="material-icons">help_outline</i> </a>
+        <a @click="$store.commit(`toggleSessionModal`, false)">
```
Comment (faboweb) [inline app/src/renderer/components/common/TmSessionHardware.vue]: is this testable?

File: app/src/renderer/components/common/TmSessionImport.vue [hunk 6]
```diff
@@ -2,15 +2,15 @@
...
-          @click="setState('welcome')"
-        >
+        <a @click="setState('welcome')">
           <i class="material-icons">arrow_back</i>
         </a>
         <div class="tm-session-title">
           Import with Seed
         </div>
-        <a @click="help"><i class="material-icons">help_outline</i></a>
+        <a @click="$store.commit(`toggleSessionModal`, false)">
```
Comment (faboweb) [inline app/src/renderer/components/common/TmSessionImport.vue]: repetitive code

File: app/src/renderer/components/common/TmSessionImport.vue [hunk 7]
```diff
@@ -187,12 +184,8 @@ export default {
   },
   mounted() {
     this.$el.querySelector(`#import-name`).focus()
-    new PerfectScrollbar(this.$el.querySelector(`.tm-session-main`))
```
Comment (faboweb) [inline app/src/renderer/components/common/TmSessionImport.vue]: lol why?

File: app/src/renderer/components/common/TmSessionHardware.vue (hunk 5 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmSessionHardware.vue]: hypothetically

Comment (faboweb) [reply app/src/renderer/components/common/TmSessionHardware.vue]: is it tested?

File: app/src/renderer/components/common/TmSessionImport.vue (hunk 6 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmSessionImport.vue]: yes. i couldn't come up with a simple way of addressing this without re-architecting the session modals / slots.

File: app/src/renderer/components/common/TmSessionImport.vue (hunk 7 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmSessionImport.vue]: it is not necessary anymore

File: app/src/renderer/components/common/TmSessionHardware.vue (hunk 5 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmSessionHardware.vue]: i just added a test

File: app/src/renderer/components/common/TmSessionImport.vue (hunk 7 above)
Comment (faboweb) [reply app/src/renderer/components/common/TmSessionImport.vue]: isn't it changing the look of the scrollbar on windows?

File: app/src/renderer/components/common/TmSessionImport.vue (hunk 6 above)
Comment (faboweb) [reply app/src/renderer/components/common/TmSessionImport.vue]: Follow up https://github.com/cosmos/voyager/issues/2155

File: app/src/renderer/components/common/TmSessionHardware.vue (hunk 5 above)
Comment (faboweb) [reply app/src/renderer/components/common/TmSessionHardware.vue]: I think creating a method would have been the cleaner solution as it is done in most of the other components.

**Comment by faboweb** [comment]
I had outstanding requests :/

File: app/src/renderer/components/common/TmSessionImport.vue (hunk 7 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmSessionImport.vue]: hmm that i'm not sure of. but we can style scrollbars in css if need be. the use of this library seems like a code smell to me ... and predicting which component will need a custom scrollbar is not scalable.

File: app/src/renderer/components/common/TmSessionHardware.vue (hunk 5 above)
Comment (jbibla) [reply app/src/renderer/components/common/TmSessionHardware.vue]: yes, you are right. please fix in #2155 🙏 🐱

**Comment by jbibla** [comment]
> I had outstanding requests

evidence that 2 approvals if needed 😄

**Pull request merged**